### ISSUE TITLE
[AWS SQS inbound] pre-release tesing on Pre-Prod #26

### ISSUE THREAD

**Issue by Oleksiivanov** [camunda/connectors#688]
Main test cases for sqs inbound connector (in my, not QA, view) : 
- check that Authentication and queue properties can use 'secrets' https://docs.camunda.io/docs/components/console/manage-clusters/manage-secrets/;
- check that polling wait time can be only >0 and <20
- if you set the attributes message name so you can see these attributes in response;
- check activation condition (with correlation keys)
- check that start event connector and intermediate connector exist and works.

To test AWS SQS inbound connector you need to have an AWS account with the required permissions. More you can find here : https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-setting-up.html

To configure the SQS inbound Connector and receive messages from your SQS Queue, follow these steps:

1. Set the relevant IAM key and secret pair in the **Authentication** section. The value can be plain text
2. In the **Queue Properties** section, set the URL of your SQS Queue and its region.
3. In the **Message polling properties** section, set the polling wait time. This is the duration (in seconds) for which the call waits for a message to arrive in the queue before returning. See the [official documentation](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-short-and-long-polling.html) for more details. can be 0-20
4. (Optional) In the **Use next attribute names for activation condition** section, set an array of **Attribute names** or **Message attribute name** (e.g., `["attributeName1", "attributeName2"]`) to receive messages from the queue with specific metadata. Learn more about message metadata [here](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-message-metadata.html).
5. (Optional) Configure the **Activation Condition**. For example, if an external message has the body `{"messageId": 1, "body": "Hi team", "messageAttributes":{"key":{"stringValue":"value"}}...}`, the **Activation Condition** value might look like `=(messageAttributes.key.stringValue="value")`. Leave this field empty to receive all messages every time.
  
<img width="349" alt="connectors-aws-sqs-start-event-activation" src="https://github.com/camunda/connectors-bundle/assets/108869886/911f2ea1-0abd-4878-a40e-1cfbdacb2f5b">

6. Set **Variable mapping**. For example, to get only the message body, you can set `{resultBody: body}` in the **Result expression**.

When using the **AWS SQS Inbound Connector** with an **Intermediate Catch Event**, fill in the **Correlation key (process)** and **Correlation key (payload)**.

- **Correlation key (process)** is a FEEL expression that defines the correlation key for the subscription. This corresponds to the **Correlation key** property of a regular **Message Intermediate Catch Event**.
- **Correlation key (payload)** is a FEEL expression used to extract the correlation key from the incoming message. This expression is evaluated in the Connector Runtime and the result is used to correlate the message.

For example, given that your correlation key is defined with `myCorrelationKey` process variable, and the request body contains `" "messageAttributes":{"key":{"stringValue":"value"}}`, your correlation key settings will look like this:

- **Correlation key (process)**: `=myCorrelationKey`
- **Correlation key (payload)**: `=messageAttributes.key.StringValue`

## Activate the SQS inbound Connector

Once you click the **Deploy** button, your SQS inbound Connector will be activated and publicly available. Whenever the SQS inbound Connector receives a new message, a new BPMN process will be created.

EPIC: https://github.com/camunda/camunda-platform-docs/pull/2096
DOCS : https://github.com/camunda/camunda-platform-docs/pull/2096

**Comment by GeethaParthasarathy** [comment]
@Oleksiivanov The initial set of test cases are here in the link:vhttps://camunda.testrail.com/index.php?/suites/view/141&group_by=cases:section_id&group_order=asc&display_deleted_cases=0&group_id=3043 Kindly let me know your feedback on the same

**Comment by GeethaParthasarathy** [comment]
@Oleksiivanov I have completed testing the SQS inbound start and intermediate connector. Overall looks good. 

- I have one suggesstion for error handling or maybe a info message for `message attribute names ` field that needs to have a "array" value. No where we could see that it must be an array , I stumbled upon not giving this value as an array and found the connector to be "DOWN" always later realized that this value needs to be an array. 
Hence it would be nice if we let the users know this upfront. Pls let me know if I could create a user improvement Issue for this.
<img width="259" alt="fce870e6203198d0ee812e79d917a000" src="https://github.com/camunda/connectors-bundle/assets/85225325/0b36e5f3-b902-42f0-8d67-106a846a5113">


- As discussed with you the documentation for Inbound SQS is still not merged and hence I see the links to the docs in the properties panel broken(which is a known issue)
